Updating default embedded mongo version to 3.6.4

bilelbm opened the pull request:
Hi,
Since Mongo 3.6 alot of new features are introduced so its more practical to use 3.6 instead of 3.5

pivotal-issuemaster commented:
@bilelbm Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=15801)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-boot&pullRequestId=15801) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

wilkinsona commented:
Thanks for the PR. We've tried to bump the version to 3.6.5 [in the past](https://github.com/spring-projects/spring-boot/commit/1a8f519922b288491cc64d5ef0a0b95f443d7a66) and had to drop back to 3.5.5. What made you choose 3.6.4 rather than a more up-to-date version in the 3.6.x line? Also, did you build the changes prior to submitting them as they are [failing on Concourse](
https://ci.spring.io/builds/57155).

bilelbm commented:
i changed the version from 3.6.4 to 3.6.5 (for me 3.6.4 was the most stable version i used), updated the function that gets the Mongo version and fixed the tests in my latest commit. It is possible now to use Mongo version 4.x too since it is supported since version 2.1 of the embed Mongo dependency.

Ps: I built the changes before committing and they passed.

bilelbm commented:
I understand that.
Thank you.
